### PR TITLE
Add Github Action tag and release workflow for Terraform module

### PR DESCRIPTION
**Pull request by begeekandchic**
This is a direct copy and paste of `github-reusable-workflows/.github/workflows/reusable-workflow-tag-and-release-terraform-module.yaml` into this repo from github-reusable-workflows, since this repo is public and originally forked from trussworks/terraform-aws-ecs-service. The Github action workflow is referenced locally from this repo, instead of `github-reusable-workflows` repo. This is because github-reusable-workflows is a private repo, and reusable workflows can only be referenced within an organization's private repos.

### CODE REVIEW

**Review by soleares — COMMENTED** [review]
Let's put this on-hold until we talk and come up with a plan since this is a public repo.

File: .github/workflows/reusable-workflow-tag-and-release-terraform-module.yaml
Comment (soleares) [inline .github/workflows/reusable-workflow-tag-and-release-terraform-module.yaml:1]: I don't see a good reason to have the workflow split between 2 files here. Let's just add the steps from this file to the main workflow file and remove this file. It's helpful to have these reusable workflows when it's shared, but it's not shared in this case.

File: .github/workflows/tag-and-release.yaml [hunk 1]
```diff
@@ -20,7 +17,10 @@ permissions:
 
 
 jobs:
-  # Call the reusable workflow.
+  # Call the reusable workflow locally
   call-workflow:
```
Comment (soleares) [inline .github/workflows/tag-and-release.yaml:21]:
```suggestion
  tag-and-release:
```

File: .github/workflows/tag-and-release.yaml [hunk 2]
```diff
@@ -20,7 +17,10 @@ permissions:
...
-    name: Call Workflow
-    uses: onemedical/github-reusable-workflows/.github/workflows/reusable-tag-and-release-terraform-module.yaml@main
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3 # Necessary to access local action
+      - name: Call tag and release terraform module workflow
+        uses: ./.github/workflows/reusable-tag-and-release-terraform-module.yaml
```
Comment (soleares) [inline .github/workflows/tag-and-release.yaml:26]:
```suggestion
      - name: 📥 Check out code
        uses: actions/checkout@v3

      - name: 🏷 Create a tag
        id: create-tag
        uses: butlerlogic/action-autotag@stable
        with:
          strategy: regex
          root: "version"
          regex_pattern: "(\\d+\\.\\d+\\.\\d+)"
          tag_prefix: "v"
        env:
          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"

      - name: 🦄 Create a release
        uses: ncipollo/release-action@v1
        with:
          generateReleaseNotes: true
          tag: ${{ steps.create-tag.outputs.tagname }}
          token: ${{ secrets.GITHUB_TOKEN }}
```

**Review by soleares — COMMENTED** [review]
I think we just need a single workflow file here. I don't think the reusable workflow makes sense for a public repo since it's not shared/reusable.

**Review by Svetlanko — COMMENTED** [review]
yeah, i think there is not much value to set up 1 reusable workflow in the public fork. same thoughts as @soleares we don't gain much by having 2 layers here. just copy/paste `.github/workflows/reusable-workflow-tag-and-release-terraform-module.yaml` file content into `.github/workflows/tag-and-release.yaml` file, remove references to reusable workflow, etc and  it should be good 🙌 

https://github.com/onemedical/crane/blob/main/.github/workflows/tag_and_release.yml this is not exactly TF module example and not a public fork but kinda a similar example

File: .github/workflows/tag-and-release.yaml [hunk 3]
```diff
@@ -1,7 +1,4 @@
-# onemedical/<repo>/.github/workflows/tag-and-release-terraform-module.yaml
-#
-# This is the workflow for distribution to repositories across the organization.
-# It will call the reusable Tag and Release Terraform Module workflow,
+# This is the workflow that calls the reusable Tag and Release Terraform Module workflow,
 # which will create a new tag based on a value in version file and create a GitHub release.
```
Comment (soleares) [inline .github/workflows/tag-and-release.yaml:2]: I would remove the "reusable" language
```suggestion
# Create a new GitHub tag and release based on a value in version file.
```

**Review by soleares — APPROVED** [review]
LGTM. Just a few suggested comment updates.

**Pull request merged**